src/screens/MainScreen/index.js

```diff
@@ -18,11 +23,20 @@ const DEFAULT_LOCATION = {
 };
 
 const MainScreen = () => {
+  const [isModalVisible, setModalVisible] = useState(false);
   const { userHasLocation, userLocation } = useUserLocation();
   const { latitude, longitude } = userLocation;
   const navigation = useNavigation();
+  const dispatch = useDispatch();
 
-  const [isModalVisible, setModalVisible] = useState(false);
+  useEffect(() => {
+    dispatch(getTargets());
+    dispatch(getTopics());
+  }, []);
+
+  const { targets } = useSelector(({ targets }) => ({
+    targets: targets?.targets ?? [],
```

Comment:
What do you think about moving the `useEffect` and this `useSelector` to a new hook called `useTargets` or something like that? The hook would call these two hooks and then return the list of targets.

Just to avoid having the logic directly here, and also to hide the implementation of _**how**_ it's done.